include/llvm-dialects/Dialect/Dialect.h

```diff
@@ -249,7 +249,9 @@ bool isSimpleOperationDecl(const llvm::Function *fn, llvm::StringRef name);
 bool isOverloadedOperationDecl(const llvm::Function *fn, llvm::StringRef name);
 
 bool isSimpleOperation(const llvm::CallInst *i, llvm::StringRef name);
+bool isSimpleOperation(llvm::StringRef fn, llvm::StringRef name);
 bool isOverloadedOperation(const llvm::CallInst *i, llvm::StringRef name);
+bool isOverloadedOperation(llvm::StringRef fn, llvm::StringRef name);
```

Comment:
I don't think you need these overloads. Instead, add an `isOperationDecl` (see above) that takes an `isOverloaded` argument in addition to the name.